core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,463 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Symbology
+ */
+
+import { compareBooleans, compareNumbers, CompressedId64Set, NonFunctionPropertiesOf, OrderedId64Iterable } from "@itwin/core-bentley";
+import { LinePixels } from "./LinePixels";
+import { RgbColor, RgbColorProps } from "./RgbColor";
+
+/** JSON representation of a [[ContourStyle]].
+ * @public
+ */
+export interface ContourStyleProps {
+  /** See [[ContourStyle.color]]. */
+  color?: RgbColorProps;
+  /** See [[ContourStyle.pixelWidth]]. */
+  pixelWidth?: number;
+  /** See [[ContourStyle.pattern]]. */
+  pattern?: LinePixels;
+}
+
+/** A type containing all of the properties of [[ContourStyle]] with none of the methods and with the `readonly` modifiers removed.
+ * Used by [[ContourStyle.create]] and [[ContourStyle.clone]].
+ * @public
+ */
+export type ContourStyleProperties = NonFunctionPropertiesOf<ContourStyle>;
+
+/** The style settings used by either a minor or major contour.
+ * @see [[Contour.majorStyle]]
+ * @see [[Contour.minorStyle]]
+ * @public
+ */
+export class ContourStyle {
+  /** The color in which to draw the contour lines. Default: black. */
+  public readonly color: RgbColor;
+  /** The width in screen pixels of the contour lines.
+   * Useful values range between 1 and 8.5, in increments of 0.5. Other values will be rounded to meet these criteria.
```

Comment:
I assumed but did not verify that rounding is what happens, please adjust if I guessed wrong.

Reply:
@DStradley please take a look when you can.

Reply:
Values outside of that range are clamped to that range, values inside that range are rounded to 0.5 increments.